src/pandantic/validators/pandas.py

```diff
@@ -0,0 +1,132 @@
+import logging
+import math
+import os
+from typing import Any, Optional
+
+import pandas as pd
+from multiprocess import (  # type:ignore # pylint: disable=no-name-in-module
+    Process,
+    Queue,
+)
+
+from pandantic.types import SchemaTypes
+from pandantic.validators.baseclass import BaseValidator
+
+
+class PandasValidator(BaseValidator):
+    def __init__(self, schema: SchemaTypes):
+        self.schema = schema
+
+    def validate(
+        self,
+        dataframe: pd.DataFrame,
+        errors: str = "raise",
+        context: Optional[
+            dict[str, Any]
+        ] = None,  # pylint: disable=consider-alternative-union-syntax,useless-suppression
+        n_jobs: int = 1,
+        verbose: bool = True,
+    ) -> pd.DataFrame:
+        """Validate a DataFrame using the schema defined in the Pydantic model.
+
+        Args:
+            dataframe (pd.DataFrame): The DataFrame to validate.
+            errors (str, optional): How to handle validation errors. Defaults to "raise".
+            context (Optional[dict[str, Any], None], optional): The context to use for validation.
+            n_jobs (int, optional): The number of processes to use for validation. Defaults to 1.
+            verbose (bool, optional): Whether to log validation errors. Defaults to True.
+
+        Returns:
+            pd.DataFrame: The DataFrame with valid rows in case of errors="filter".
+        """
+        errors_index = []
+        logging.debug("Amount of available cores: %s", os.cpu_count())
+
+        dataframe = dataframe.copy()
+        dataframe["_index"] = dataframe.index
+
+        if n_jobs != 1:
+            if n_jobs < 0:
+                n_jobs = os.cpu_count()  # type: ignore
+
+            chunks = []
+            chunk_size = math.floor(len(dataframe) / n_jobs)
+            num_chunks = len(dataframe) // chunk_size + 1
+
+            q = Queue()
+
+            for i in range(num_chunks):
+                chunks.append(dataframe.iloc[i * chunk_size : (i + 1) * chunk_size])
+
+            for i in range(num_chunks):
+                p = Process(target=self._validate_row, args=(chunks[i], q), daemon=True)
+                p.start()
+
+            num_stops = 0
+            for i in range(num_chunks):
+                while True:
+                    index = q.get()
+                    if index is None:
+                        num_stops += 1
+                        break
+
+                    errors_index.append(index)
```

Comment:
**This logic is incorrect!** The invalid chunks get messaged back out as invalid rows, which then will remove too many rows if `errors="filter"`.

The easiest fix is probably changing the `_validate_chunks` function to return errors row indices, then re-combing these lists.

Reply:
Are you sure, only the validation error rows are being put into the queue and whenever the chunk is finished, a None is put onto to the Queue indicating the batch is done. I did change the name of the function though.

Reply:
Ah I see, yes that makes sense, I was misunderstanding the use of the queue.